### PR TITLE
fix(components/lookup): autocomplete position in modal with omnibar

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2698938](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2698938)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1799/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/1799/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1799/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1799/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1799/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1799/playground/)

File: apps/integration/src/app/integrations/lookup-in-modal/modal-lookup.component.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
```
Comment (Blackbaud-SteveBrush) [inline apps/integration/src/app/integrations/lookup-in-modal/modal-lookup.component.ts]: Is there a test for this component? The "integrations" app is a place to put unit tests for complex component interactions. If the intent is to have an example that can be interacted with manually, we should put this in the playground app.

File: libs/components/core/src/lib/modules/affix/dom-utils.ts [hunk 2]
```diff
@@ -65,12 +65,14 @@ export function getOverflowParents(child: HTMLElement): HTMLElement[] {
...
       overflowY === 'scroll'
     ) {
-      results.push(parentElement);
+      results.unshift(parentElement);
     }
 
     parentElement = parentElement.parentNode;
   }
 
+  results.unshift(bodyElement);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/core/src/lib/modules/affix/dom-utils.ts]: Using `Array.unshift` is generally discouraged due to performance reasons. What's the purpose of this change?

Comment (johnhwhite) [reply libs/components/core/src/lib/modules/affix/dom-utils.ts]: The stack order was incorrect. It was expected that the first item was `<body>` and the last item was the closest parent with overflow, but prior to this change, the first item was body, the second item was the closest overflow parent, and the last item was the highest overflow parent within `<body>` because parents were pushed to the end of the array as it walked up the tree.

I can flip the stack to have body last if that's preferable, and then adjust where it's used accordingly.

File: apps/integration/src/app/integrations/lookup-in-modal/modal-lookup.component.ts (hunk 1 above)
Comment (johnhwhite) [reply apps/integration/src/app/integrations/lookup-in-modal/modal-lookup.component.ts]: This is in the integration app to support the respective e2e test. I can add a unit test for this component.

Comment (Blackbaud-SteveBrush) [reply apps/integration/src/app/integrations/lookup-in-modal/modal-lookup.component.ts]: I didn't realize we had an "integration" E2E application. Thanks for creating the unit test! 👍

**Review by Blackbaud-SteveBrush — DISMISSED** [review]

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/1801
